### ISSUE TITLE
 Sharing selection dialog is not displayed correctly in Electron

### ISSUE THREAD

**Issue by kobayashi-winc** [jitsi/jitsi-meet#8201]
I'll give you a fix for this issue.

Consider reflecting it in your code.

function.js

export function obtainDesktopSources(types, options = {}) {  
    
    const capturerOptions = {
        types
    };   
    
    if (options.thumbnailSize) {
        capturerOptions.thumbnailSize = options.thumbnailSize;
    }   
    
    return new Promise((resolve, reject) => {  
        const JitsiMeetElectron = {
                    obtainDesktopStreams(callback, errorCallback, options = {}) {
                    var electron = require('electron');
                    electron.desktopCapturer.getSources(options).then(sources => {
                    callback(sources);
                }).catch(error => {
                    errorCallback(error);
                });
            }
        };

        if (JitsiMeetElectron && JitsiMeetElectron.obtainDesktopStreams) {
            JitsiMeetElectron.obtainDesktopStreams(
                sources => resolve(_seperateSourcesByType(sources)),
                error => {
                    logger.error(
                        `Error while obtaining desktop sources: ${error}`);
                    reject(error);
                },
                capturerOptions
            );
        } else {
            const reason = 'Called JitsiMeetElectron.obtainDesktopStreams'
                + ' but it is not defined';

            logger.error(reason);

            return Promise.reject(new Error(reason));
        }
    });
}

**Comment by saghul** [comment]
Can you please elaborate? I can't spot the problem.

**Comment by kobayashi-winc** [comment]
I'm worried if the meaning can be understood because I'm converting Japanese with translation software.

When developing my own Jitsi Meet that runs on Electron, there was a problem that the sharing dialog was not displayed correctly.

I parsed app.bundle.min.js and found that the cause of the error was the unimplemented obtainDesktopStreams method.

Perhaps this method is the correct way to capture it in some special way, but I didn't know how to do it, so I wrote the code directly in app.bundle.min.js to handle it.

Right now I'm trying to fix the source I got from gitHub and do make.
However, I got a make error with require ('electron').

![app bundle min js](https://user-images.githubusercontent.com/69136389/100988371-ffeab000-3592-11eb-99b0-b09e088bfb7d.png)
